fs/operations.go

```diff
@@ -1303,3 +1364,24 @@ func MoveFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err err
 func CopyFile(fdst Fs, fsrc Fs, dstFileName string, srcFileName string) (err error) {
 	return moveOrCopyFile(fdst, fsrc, dstFileName, srcFileName, true)
 }
+
+func (j jsonPrinter) Start(w io.Writer) {
+	syncFprintf(w, "[")
+}
+
+func (j jsonPrinter) End(w io.Writer) {
+	syncFprintf(w, "]")
+}
+
+func (j *jsonPrinter) Fprintf(w io.Writer, skipSeperator bool, format string, a ...interface{}) {
+	// Lets skip the seperator when calling the function itself
+	if j.seperatorRequired && !skipSeperator {
+		j.Fprintf(w, true, ",")
+	}
+	// Set the seperator before locking so parallel things don't get in the way of the lock
+	j.seperatorRequired = true
+
+	j.mu.Lock()
```

Comment:
After some testing, if I put the lock call at the very top of the method, rclone locks up. But if I put it as shown below, rclone runs fine.

```// Lets skip the seperator when calling the function itself
if j.seperatorRequired && !skipSeperator {
	j.Fprintf(w, true, ",")
}
j.mu.Lock()
defer j.mu.Unlock()
// Set the seperator before locking so parallel things don't get in the way of the lock
j.seperatorRequired = true
_, _ = fmt.Fprintf(w, format, a...)```